docs: update the readthedocs integration for PRs in this repo

wmamills opened the pull request:
Upstream Read The Docs has tightened up the schema checking and this broke what we were doing for PRs.

A good number of things were tried but the best compromise is to:
* supply a dummy sphinx config file in this repo
* override the default build command to use the openamp-docs dir

Also:
* add more documentation about what is going on
* lock to html only format.  PRs only do html anyway but make sure

wmamills commented:
@arnopo @tnmysh I am assuming that the docs that are now in open-amp/doc/apps will be moved here to docs/apps.
If this is not true it is a bit silly to create the doc dir only to hold a do nothing file.
Let me know what you think

tnmysh commented:
Should we open PR to track that migration?

wmamills commented:
@tnmysh yes, please open an issue
@arnopo do you both agree that this is the right thing to do?  Move them as is from open-amp to openamp-system-reference?
The other alternative is to move the directly to openamp-docs.

tnmysh commented:
> @tnmysh yes, please open an issue in open-amp @arnopo do you both agree that this is the right thing to do? Move them as is from open-amp to openamp-system-reference? The other alternative is to move the directly to openamp-docs.

I think keeping apps related docs in system-reference will be more helpful. Then openamp-docs can point link to this repo. But I am open to otherway as well.

arnopo commented:
> @tnmysh yes, please open an issue in open-amp @arnopo do you both agree that this is the right thing to do? Move them as is from open-amp to openamp-system-reference? The other alternative is to move the directly to openamp-docs.

I would be in favor of having complex documentation (for instance, PNG schematics) in the openamp-docs repo and just a README in the system-reference demo explaining how to build and run the demo.

And regarding @sipke's work on documentation, we probably have some duplications that should be cleaned up in the openamp-docs repo.

tnmysh commented:
open-amp issue is here: https://github.com/OpenAMP/open-amp/issues/638

wmamills commented:
Based on the discussion at OpenAMP system reference meeting on Feb 19th, we don't expect the doc/ directory in this repo to conatin anything.  I don't think we should create it just for this one file.

I suggest I rework this PR to move the doc/readthedocs_conf.py file to /.readthedocs_conf.py (in the root dir and with a leading dot)

@arnopo when you get back from vacation let me know if you agree

wmamills commented:
@arnopo I have moved the config doc to the root dir (with a leading . ) and I don't create the doc dir.
I have tested this in my own account and it works OK.
I have removed the RFC status on this.
